subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/IndexPageGenerator.java

```diff
@@ -16,96 +16,165 @@
-                    div().id("controls").end();
-                    for (String testName : testNames) {
-                        PerformanceTestHistory testHistory = store.getTestResults(testName, 5, 14, ResultsStoreHelper.determineChannel());
-                        List<? extends PerformanceTestExecution> results = testHistory.getExecutions();
-                        results = filterForRequestedCommit(results);
-                        if (results.isEmpty()) {
-                            archived.put(urlEncode(testHistory.getId()), testHistory.getDisplayName());
-                            continue;
-                        }
-                        h2().classAttr("test-execution");
-                            text("Test: " + testName);
+            }
+
+            private void renderHeaders() {
+                tr();
+                    th().text("Scenario").end();
+                    th().colspan("2").text("Control Group").end();
+                    th().colspan("2").text("Experiment Group").end();
+                    th().colspan("2").text("Regression").end();
+                end();
+            }
+
+            private String determineScenarioCss(ScenarioBuildResultData scenario) {
+                if(scenario.isSuccessful()) {
+                    return "success";
+                } else if(scenario.isRegressed()) {
+                    return "warning";
+                } else {
+                    return "danger";
+                }
+            }
+
+            private void renderScenario(ScenarioBuildResultData scenario) {
+                if (scenario.getExperimentData().isEmpty()) {
+                    renderScenario(scenario, 0);
+                } else {
+                    for (int i = 0; i < scenario.getExperimentData().size(); i++) {
+                        renderScenario(scenario, i);
+                    }
+                }
+            }
+
+            private void renderScenario(ScenarioBuildResultData scenario, int experimentIndex) {
+                tr().classAttr(determineScenarioCss(scenario));
+
+                if (experimentIndex == 0) {
+                    td().rowspan(scenario.getExperimentData().isEmpty() ? "1" : "" + scenario.getExperimentData().size());
+                        a().href(scenario.getWebUrl()).classAttr("label label-" + determineScenarioCss(scenario));
+                            big().u().text(scenario.getScenarioName()).end().end();
                         end();
-                        table().classAttr("history");
-                        tr().classAttr("control-groups");
-                            th().colspan("2").end();
-                            th().colspan(String.valueOf(testHistory.getScenarioCount() * getColumnsForSamples())).text("Average execution time").end();
+                        a().href("tests/" + urlEncode(scenario.getScenarioName().replaceAll("\\s+", "-")) + ".html").classAttr("label label-info");
```

Comment:
I don't really feel like the `label` classes add much to the whole layout. I think it would be cleaner just having the link without any special styling. I think the background of the row is already enough.

Reply:
That's what it would look like:

![image](https://user-images.githubusercontent.com/423186/45186699-afa70b00-b22e-11e8-827c-87498d47a725.png)

I think the two links (`details...`, `build`) should go to separate columns. WDYT?